iradix.go

```diff
@@ -357,6 +383,61 @@ func (t *Txn) delete(parent, n *Node, search []byte) (*Node, *leafNode) {
 	return nc, leaf
 }
 
+// delete does a recursive deletion
+func (t *Txn) deletePrefix(parent, n *Node, search []byte) (*Node, int) {
+	// Check for key exhaustion
+	if len(search) == 0 {
+		// Remove the leaf node
+		nc := t.writeNode(n, true)
+		subTreeSize := 0 //count this node
+		//recursively walk from all edges of the node to be deleted, tracking their mutate channels in the transaction
+		for _, e := range n.edges {
+			t.recursiveWatchNotify(e.node, func(n *Node) bool {
+				subTreeSize++
+				return false
+			})
+		}
+		if n.isLeaf() {
+			nc.leaf = nil
+			subTreeSize += 1
+		}
+		nc.edges = nil // deletes the entire subtree
+
+		return nc, subTreeSize
+	}
+
+	// Look for an edge
+	label := search[0]
+	idx, child := n.getEdge(label)
+	if child == nil || (!bytes.HasPrefix(child.prefix, search) && !bytes.HasPrefix(search, child.prefix)) {
```

Comment:
The first condition handles the case when the prefix being deleted does not have an associated leaf node. E.g when you want to delete the prefix `"AB"`, in a tree that contains `"ABCDE"` and `"ABCD"`, then at this point child.prefix will be `"ABCDE"` and will contain `"AB"` as its prefix.  

I'll add some more test cases, i want to make sure i didn't miss other edge cases here.